framework/src/base/MooseError.C

```diff
@@ -0,0 +1,26 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MooseError.h"
+
+namespace Moose {
+
+void MPIAbortWrapper()
+{
+#if defined(LIBMESH_HAVE_MPI)
+  MPI_Abort(libMesh::GLOBAL_COMM_WORLD, 1);
+#endif
```

Comment:
I could make the wrapper a macro that is wrapped in an ifdef. Yeah yeah, you need to go deeper.. Anyhow that would result in less code duplication and I could put the std:;abort in the proper place.